app/models/provider.rb

```diff
@@ -4,16 +4,20 @@ class Provider < ApplicationRecord
   has_many :users
   has_many :trainees
   has_many :apply_applications
-  has_many :courses
 
   validates :name, presence: true
   validates :dttp_id, uniqueness: true, format: { with: /\A[a-f0-9]{8}-([a-f0-9]{4}-){3}[a-f0-9]{12}\z/i }
   validates :code, format: { with: /\A[A-Z0-9]+\z/i }, allow_blank: true
 
+  audited
+
+  has_associated_audits
+
   def code=(cde)
     self[:code] = cde.to_s.upcase
   end
 
-  audited
-  has_associated_audits
+  def courses
```

Comment:
I think the relationship would be better as it will be chainable. I'd still go with not joining them on the non-standard foreign key but happy to approve this if you prefer it. I don't have a particularly compelling argument for doing it 'my' way.